bot/cogs/slowmode.py

```
@@ -0,0 +1,86 @@
+    async def slowmode_group(self, ctx: Context) -> None:
+        """Get and set the slowmode delay for a given text channel."""
+        await ctx.send_help(ctx.command)
+
+    @slowmode_group.command(name='get', aliases=['g'])
+    async def get_slowmode(self, ctx: Context, channel: TextChannel) -> None:
+        """Get the slowmode delay for a given text channel."""
+        delay = relativedelta(seconds=channel.slowmode_delay)
+
+        # Say "0 seconds" instead of "less than a second"
+        if channel.slowmode_delay == 0:
+            humanized_delay = '0 seconds'
+        else:
+            humanized_delay = time.humanize_delta(delay)
+
+        await ctx.send(f'The slowmode delay for {channel.mention} is {humanized_delay}.')
+
+    @slowmode_group.command(name='set', aliases=['s'])
+    async def set_slowmode(self, ctx: Context, channel: TextChannel, delay: DurationDelta) -> None:
+        """Set the slowmode delay for a given text channel."""
+        # Convert `dateutil.relativedelta.relativedelta` to `datetime.timedelta`
+        # Must do this to get the delta in a particular unit of time
+        utcnow = datetime.utcnow()
+        slowmode_delay = (utcnow + delay - utcnow).total_seconds()
+
+        humanized_delay = time.humanize_delta(delay)
+
+        if 0 <= slowmode_delay <= 21600:
+            log.info(f'{ctx.author} set the slowmode delay for #{channel} to {humanized_delay}.')
+
+            await channel.edit(slowmode_delay=slowmode_delay)
+            await ctx.send(
+                f'{Emojis.check_mark} The slowmode delay for {channel.mention} is now {humanized_delay}.'
+            )
+
+        else:
+            log.info(
+                f'{ctx.author} tried to set the slowmode delay of #{channel} to {humanized_delay}, '
+                'which is not between 0 and 6 hours.'
+            )
+
+            await ctx.send(
+                f'{Emojis.cross_mark} The slowmode delay must be between 0 and 6 hours.'
+            )
```

This could be moved up as a fail early check that returns, instead of if/else

---

I think that it would be clearer to leave as an if/else in this situation. Does anyone else have thoughts on this?

---

ah, yeah you're right. greedy would make sense if you wanted multiple, which doesn't make sense. Sorry, it was late.